packages/subcommands/src/lib/Subcommand.ts

```diff
@@ -88,16 +89,44 @@ export class SubCommandPluginCommand extends Command {
 		const result = await fromAsync(async () => {
 			interaction.client.emit(Events.ChatInputSubcommandRun, interaction, subcommand, payload);
 			let result: unknown;
+			subcommand.type ??= 'method';
+
+			if (subcommand.type === 'command') {
+				const command = this.container.stores.get('commands').get(subcommand.name);
+				if (command && command.chatInputRun) {
+					// Run global preconditions:
+					const globalResult = await this.container.stores
+						.get('preconditions')
+						.chatInputRun(interaction, command as ChatInputCommand, context as any);
+					if (!globalResult.success) {
+						this.container.client.emit(SapphireEvents.ChatInputCommandDenied, globalResult.error, payload);
+						return;
+					}
+
+					// Run command-specific preconditions:
+					const localResult = await command.preconditions.chatInputRun(interaction, command as ChatInputCommand, context as any);
+					if (!localResult.success) {
+						this.container.client.emit(SapphireEvents.ChatInputCommandDenied, localResult.error, payload);
```

Comment:
Since the plugin now has its own events for run, error, etc. maybe it'd be worth making these ChatInputSubcommandDenied and MessageSubcommandDenied for message commands?

Reply:
Done